src/test/fixtures/profiles/gecko-profile.js

```diff
@@ -711,3 +717,98 @@ export function createGeckoCounter(thread: GeckoThread): GeckoCounter {
   }
   return geckoCounter;
 }
+
+export function createGeckoProfilerOverhead(
+  thread: GeckoThread
+): GeckoProfilerOverhead {
+  // Helper class to calculate statistics.
+  class ProfilerStats {
+    n = 0;
+    sum = 0;
+    min = Number.MAX_SAFE_INTEGER;
+    max = 0;
+    count(val: number) {
+      ++this.n;
+      this.sum += val;
+      if (val < this.min) {
+        this.min = val;
+      }
+      if (val > this.max) {
+        this.max = val;
+      }
+    }
+  }
+
+  const intervals = new ProfilerStats();
+  const overheads = new ProfilerStats();
+  const lockings = new ProfilerStats();
+  const cleanings = new ProfilerStats();
+  const counters = new ProfilerStats();
+  const threads = new ProfilerStats();
+
+  // Fill the profiler overhead data.
+  const data = [];
+  for (let i = 0; i < thread.samples.data.length; i++) {
+    // Go through all the thread samples and create a corresponding counter entry.
+    const time = thread.samples.data[i][1] * 1000;
+    const prevTime = i === 0 ? 0 : thread.samples.data[i - 1][1] * 1000;
+    // Create some arbitrary (positive integer) values for the individual overheads.
+    const lockingTime = Math.floor(50 * Math.sin(i) + 50);
+    const cleaningTime = Math.floor(50 * Math.sin(i) + 55);
+    const counterTime = Math.floor(50 * Math.sin(i) + 60);
+    const threadTime = Math.floor(50 * Math.sin(i) + 65);
+    const interval = time - prevTime;
+
+    intervals.count(interval);
+    lockings.count(lockingTime);
+    cleanings.count(cleaningTime);
+    counters.count(counterTime);
+    threads.count(threadTime);
+    overheads.count(lockingTime + cleaningTime + counterTime + threadTime);
+
+    data.push([time, lockingTime, cleaningTime, counterTime, threadTime]);
+  }
+
+  const profiledDuration =
+    thread.samples.data[thread.samples.data.length - 1][1] -
+    thread.samples.data[0][1];
+  const statistics = {
```

Comment:
Flow will probably give nicer errors if this has an inline definition. Otherwise it relies on the type checking of the function signature and return value, which can be a bit confusing.

Reply:
Yeah, you're right. Added it, thanks!